src-plugins/medBinaryOperation/medBinaryOperatorBase.cpp

```diff
@@ -0,0 +1,251 @@
+        else if ( id == "itkDataImageUShort3" )
+        {
+            res = runProcess< ImageType, itk::Image <unsigned short,3> >();
+        }
+        else if ( id == "itkDataImageInt3" )
+        {
+            res = runProcess< ImageType, itk::Image <int,3> >();
+        }
+        else if ( id == "itkDataImageUInt3" )
+        {
+            res = runProcess< ImageType, itk::Image <unsigned int,3> >();
+        }
+        else if ( id == "itkDataImageLong3" )
+        {
+            res = runProcess< ImageType, itk::Image <long,3> >();
+        }
+        else if ( id== "itkDataImageULong3" )
+        {
+            res = runProcess< ImageType, itk::Image <unsigned long,3> >();
+        }
+        else if ( id == "itkDataImageFloat3" )
+        {
+            res = runProcess< ImageType, itk::Image <float,3> >();
+        }
+        else if ( id == "itkDataImageDouble3" )
+        {
+            res = runProcess< ImageType, itk::Image <double,3> >();
+        }
+        else
+        {
+            res = medAbstractProcess::PIXEL_TYPE;
+        }
+    }
+    return res;
+}
+
+template <class ImageType, class ImageType2> int medBinaryOperatorBase::runProcess()
+{
+    if ( !m_inputA->data() || !m_inputB->data())
+    {
+        return DTK_FAILURE;
+    }
+
+    typedef itk::Image<unsigned char, 3> ImageTypeOutput;
+    ImageTypeOutput::Pointer imageA;
+    ImageTypeOutput::Pointer imageB;
+
+    if (m_inputA->identifier() != "itkDataImageUChar3")
+    {
+        typename ImageType::Pointer imA = dynamic_cast< ImageType*>((itk::Object*)(m_inputA->data()));
+        typedef itk::CastImageFilter< ImageType, ImageTypeOutput > CasterType;
+        typename CasterType::Pointer caster = CasterType::New();
+        caster->SetInput(imA);
+        caster->Update();
+        imageA = caster->GetOutput();
+    }
+    else
+    {
+        imageA = dynamic_cast< ImageTypeOutput*>((itk::Object*)(m_inputA->data()));
+    }
+
+    if (m_inputB->identifier() != "itkDataImageUChar3")
+    {
+        typename ImageType2::Pointer imB = dynamic_cast< ImageType2*>((itk::Object*)(m_inputB->data()));
+        typedef itk::CastImageFilter< ImageType2, ImageTypeOutput > CasterType;
+        typename CasterType::Pointer caster = CasterType::New();
+        caster->SetInput(imB);
+        caster->Update();
+        imageB = caster->GetOutput();
+    }
+    else
+    {
+        imageB = dynamic_cast< ImageTypeOutput*>((itk::Object*)(m_inputB->data()));
+    }
+    
+    if (imageA->GetLargestPossibleRegion().GetSize() != imageB->GetLargestPossibleRegion().GetSize())
+    {
+        return medAbstractProcess::DATA_SIZE;
+    }
+
+    typedef itk::InPlaceImageFilter< ImageTypeOutput, ImageTypeOutput >  FilterType;
+    typename FilterType::Pointer filter;
+    
+    if(description() == "AND")
+    {
+        typedef itk::AndImageFilter <ImageTypeOutput, ImageTypeOutput, ImageTypeOutput> AndImageFilterType;
+        filter = AndImageFilterType::New();
+    }
+
```

Comment:
Putting spaces like this between the if and the else leaves the door open for future problems. Someone might insert and if here without realising there is an else after.